src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Utilities/TypeStyle/CSharpUseImplicitTypeHelper.cs

```diff
@@ -286,6 +286,16 @@ private static bool IsSafeToSwitchToVarWithoutNeedingSpeculation(DeclarationExpr
                 return false;
             }
 
+            // var cannot be used with target typed new
+#if CODE_STYLE
```

Comment:
I have no idea if this is the right thing to do with `CODE_STYLE`, or really what `CODE_STYLE` is or means, but its Friday afternoon for me so thought I'd just put up a PR and see what happens :D

Reply:
Looks good to me. @sharwell for another pair of eyes.

Reply:
There is no need for the condition here. Just use `SyntaxKindEx` on both paths.

Reply:
➡️ #46489